ast/src/parsed/mod.rs

```diff
@@ -914,6 +914,111 @@ impl Pattern {
             Pattern::Tuple(p) => p.iter().all(|p| p.is_irrefutable()),
         }
     }
+
+    pub fn is_fully_catch_all(&self) -> bool {
```

Comment:
What's the difference to `is_irrefutable`?

Reply:
I think theoretically there should be no difference, but why do you only consider irrefutable an array with ellipsis? The case `[_,_,_]` (where all cases are catchalls) isn't irrefutable as well?

Reply:
Oh, I see, you could use it to match the array size, my bad:

    let vec = vec![1, 2, 3];

    match vec.as_slice() {
        [_,_] => println!("2!"),
        [_, _, _] => println!("3"),
        _ => println!("Other"),
    }

Reply:
Fixed. Thanks!